### PR TITLE
Fix crash when rotating a deleted shape

### PR DESCRIPTION
**Pull request by TodePond**
This PR fixes a crash when rotating a deleted shapes, and adds a unit test for it.

It also updates the docs of an editor method to communicate a current limitation that we left a 'todo' for. See code comments for more info.

### Change Type

- [x] `patch` — Bug fix

[^1]: publishes a `patch` release, for devDependencies use `internal`
[^2]: will not publish a new version

### Test Plan

1. Get two devices out for this! (or write a little console script to do it for you)
2. Rotate a shape on one device.
3. While rotating it, delete it on the other device.
4. Make sure the page doesn't crash!

- [x] Unit Tests
- [ ] End to end tests

### Release Notes

- Fixed a crash when trying to rotate a deleted shape.

### CODE REVIEW

File: packages/editor/src/lib/editor/Editor.ts [hunk 1]
```diff
@@ -5628,11 +5628,12 @@ export class Editor extends EventEmitter<TLEventMap> {
...
 	/**
 	 * Rotate shapes by a delta in radians.
+	 * Note: Currently, this assumes that the shapes are your currently selected shapes.
 	 *
 	 * @example
 	 * ```ts
-	 * editor.rotateShapesBy(['box1', 'box2'], Math.PI)
-	 * editor.rotateShapesBy(['box1', 'box2'], Math.PI / 2)
+	 * editor.rotateShapesBy(editor.selectedIds, Math.PI)
+	 * editor.rotateShapesBy(editor.selectedIds, Math.PI / 2)
```
Comment (TodePond) [inline packages/editor/src/lib/editor/Editor.ts:5636]: This is a current limitation of this method. I've added a note for now.

File: packages/editor/src/lib/test/tools/rotating.test.ts [hunk 2]
```diff
@@ -273,6 +273,34 @@ describe('When rotating...', () => {
...
+	it.skip("rotates shapes that aren't the currently selected ones", () => {
+		editor.select(ids.box1)
+
+		editor.rotateShapesBy([ids.box2], Math.PI * 0.5)
+
+		editor.expectShapeToMatch(
+			{ id: ids.box1, rotation: 0 },
+			{ id: ids.box2, rotation: Math.PI * 0.5 }
+		)
+	})
```
Comment (TodePond) [inline packages/editor/src/lib/test/tools/rotating.test.ts:303]: Eventually, we should remove the limitation. This test should pass when it's resolved!

I want to focus on docs for now, so I'll leave-as-is for the moment. But maybe we should do this now with this change as well???

File: packages/editor/src/lib/utils/rotation.ts [hunk 3]
```diff
@@ -14,12 +14,16 @@ export function getRotationSnapshot({ editor }: { editor: Editor }) {
...
+
+	// Return null if there are no selected shapes
+	if (!selectionPageCenter) {
+		return null
+	}
 
 	return {
-		selectionPageCenter: selectionPageCenter!,
-		initialCursorAngle: selectionPageCenter!.angle(originPagePoint),
+		selectionPageCenter: selectionPageCenter,
```
Comment (steveruizok) [inline packages/editor/src/lib/utils/rotation.ts:25]: We should compute this based on the shapes being passed in rather than by the selection.

Comment (TodePond) [reply packages/editor/src/lib/utils/rotation.ts:25]: Yes, see my note - this PR just prevents the crash, but we should also resolve the todo within this file. Would you rather we just do that now instead?

File: packages/editor/src/lib/utils/rotation.ts [hunk 4]
```diff
@@ -14,12 +14,16 @@ export function getRotationSnapshot({ editor }: { editor: Editor }) {
 
 	// todo: this assumes we're rotating the selected shapes
 	// if we try to rotate shapes that aren't selected, this
-	// will produce the wrong results if there are other shapes
-	// selected or else break if there are none.
```
Comment (TodePond) [inline packages/editor/src/lib/utils/rotation.ts:18]: This is the existing todo I'm referring to

**Comment by steveruizok** [comment]
Ok, let's land it.

**Pull request merged**